dev-libs/efl: bug fix & EAPI bump for 1.20.7

juippis opened the pull request:
I've been testing this for few days with different USE flag combinations and I havent had a build breakage in a while. I don't want to remove 1.20.7 from the tree yet so users who experience breakages can go back to it. I'm sure with a file and update this big, there's something I didn't notice ;)

- bump to EAPI6 because the util eclasses dont support 7 yet
- disabled tgv because its broken and doesnt compile
- enabled gl-drm
- made more sense to opengl engine switch
- removed enlightenment.eclass, and added use flags & functions used by it
- removed blockers from ancient times
- sorted RDEPEND and made it easier for eyes overall

This ebuild shall be the base for when efl-1.21 hits and I'd like to get it into tree for testing. 


Closes: https://bugs.gentoo.org/571532
Closes: https://bugs.gentoo.org/658602
Closes: https://bugs.gentoo.org/651688
Thanks to David Davidson for fix in #651688.

Package-Manager: Portage[mgorny]-2.3.36.1

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+
+	V=1 emake install DESTDIR="${D}" || die "Installing EFL files failed."
+
+	if ! use static-libs ; then
```
you can most likely delete `*.la` unconditionally.

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+
+src_configure() {
+	if use ssl && use gnutls ; then
+		einfo "You enabled both USE=ssl and USE=gnutls, but only one can be used;"
```
This is contradictory since way above we find `REQUIRED_USE="gnutls? ( ssl )"`

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+	sys-libs/zlib
+	virtual/jpeg:0=
+
+	!media-libs/elementary
```
blockers should not end up in `DEPEND`. Please make this var be `COMMON_DEPEND` and split out the blockers in a separate `RDEPEND` definition below.

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+	tslib? ( x11-libs/tslib )
+	unwind? ( sys-libs/libunwind )
+	valgrind? ( dev-util/valgrind )
+	vlc? ( >=media-video/vlc-3.0.0 )
```
this might need a slot operator

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+	physics? ( sci-physics/bullet )
+	pixman? ( x11-libs/pixman )
+	postscript? ( app-text/libspectre )
+	media-libs/libpng:0=
```
optional dependencies below unconditional, seems like you missed that line.

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+	!luajit? ( dev-lang/lua:* )
+	nls? ( sys-devel/gettext )
+	pdf? ( app-text/poppler[cxx] )
+	physics? ( sci-physics/bullet )
```
check linking for slot op necessity

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+		media-libs/gstreamer:1.0
+		media-libs/gst-plugins-base:1.0
+	)
+	harfbuzz? ( media-libs/harfbuzz )
```
check linking for slot op necessity

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+	harfbuzz? ( media-libs/harfbuzz )
+	hyphen? ( dev-libs/hyphen )
+	ibus? ( app-i18n/ibus )
+	jpeg2k? ( media-libs/openjpeg:0 )
```
check linking for slot op necessity

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+	hyphen? ( dev-libs/hyphen )
+	ibus? ( app-i18n/ibus )
+	jpeg2k? ( media-libs/openjpeg:0 )
+	libuv? ( dev-libs/libuv )
```
check linking for slot op necessity

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+	luajit? ( dev-lang/luajit )
+	!luajit? ( dev-lang/lua:* )
+	nls? ( sys-devel/gettext )
+	pdf? ( app-text/poppler[cxx] )
```
needs slot op in case it links to `libpoppler.so`

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+	pulseaudio? ( sound )
+	sdl? ( opengl )
+	vnc? ( X fbcon )
+	wayland? ( egl gles !opengl )
```
this is contradicting above `opengl` condition.

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+	postscript? ( app-text/libspectre )
+	media-libs/libpng:0=
+	pulseaudio? ( media-sound/pulseaudio )
+	raw? ( media-libs/libraw )
```
check linking for slot op necessity

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+		media-libs/mesa[gles2,wayland]
+		x11-libs/libxkbcommon
+	)
+	webp? ( media-libs/libwebp )
```
almost certainly needs slot op

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,320 @@
+
+	sys-apps/dbus
+	sys-apps/util-linux
+	sys-libs/zlib
```
please add slot op

juippis commented:
Thanks for the feedback! I removed 'libressl' since it's not supported and made some sense to gnutls<->ssl check. 

Removed .la files unconditionally. 

Added a 2nd RDEPEND to include soft blocks from COMMON_DEPEND. I hope I got it right?

Tried to make more sense to openg-egl-gles-wayland checks. and simplified it.

Moved undonditional deps before use-flag ones.

Added := slot operators for every package I could confirm needs it. Also switched openssl from :0= to :0 since efl-1.20 works with 1.1.

```diff
15c15
< IUSE="avahi +bmp dds connman debug doc drm +eet egl examples fbcon +fontconfig fribidi gif gles glib gnutls gstreamer harfbuzz hyphen +ico ibus ivi jpeg2k libressl libuv luajit neon nls opengl ssl pdf physics pixman postscript +ppm +psd pulseaudio raw scim sdl sound static-libs svg systemd tga tiff tslib unwind v4l valgrind vlc vnc wayland webp X xcf xim xine xpresent xpm"
---
> IUSE="avahi +bmp dds connman debug doc drm +eet egl examples fbcon +fontconfig fribidi gif gles glib gnutls gstreamer harfbuzz hyphen +ico ibus ivi jpeg2k libuv luajit neon nls opengl ssl pdf physics pixman postscript +ppm +psd pulseaudio raw scim sdl sound static-libs svg systemd tga tiff tslib unwind v4l valgrind vlc vnc wayland webp X xcf xim xine xpresent xpm"
17a18,20
> 	?? ( gnutls ssl )
> 	?? ( opengl egl )
> 	?? ( opengl gles )
24d26
< 	gnutls? ( ssl )
26,30c28
< 	libressl? ( ssl )
< 	opengl? (
< 		|| ( X sdl wayland )
< 		!egl
< 	)
---
> 	opengl? ( X )
37a36,41
> COMMON_DEPEND="
> 	!media-libs/elementary
> 	!media-plugins/emotion_generic_players
> 	!media-plugins/evas_generic_loaders
> "
> 
38a43,48
> 	media-libs/libpng:0=
> 	sys-apps/dbus
> 	sys-apps/util-linux
> 	sys-libs/zlib:=
> 	virtual/jpeg:0=
> 
51d60
< 	gles? ( media-libs/mesa[gles2] )
52a62,63
> 	gles? ( media-libs/mesa[gles2] )
> 	gnutls? ( net-libs/gnutls:= )
60c71
< 	jpeg2k? ( media-libs/openjpeg:0 )
---
> 	jpeg2k? ( media-libs/openjpeg:0= )
62c73
< 	luajit? ( dev-lang/luajit )
---
> 	luajit? ( dev-lang/luajit:= )
65,66c76,77
< 	pdf? ( app-text/poppler[cxx] )
< 	physics? ( sci-physics/bullet )
---
> 	pdf? ( app-text/poppler:=[cxx] )
> 	physics? ( sci-physics/bullet:= )
69d79
< 	media-libs/libpng:0=
71c81
< 	raw? ( media-libs/libraw )
---
> 	raw? ( media-libs/libraw:= )
78,84c88
< 	ssl? (
< 		gnutls? ( net-libs/gnutls )
< 		!gnutls? (
< 			!libressl? ( dev-libs/openssl:0= )
< 			libressl? ( dev-libs/libressl:0= )
< 		)
< 	)
---
> 	ssl? ( dev-libs/openssl:= )
91c95
< 	tslib? ( x11-libs/tslib )
---
> 	tslib? ( x11-libs/tslib:= )
101c105
< 	webp? ( media-libs/libwebp )
---
> 	webp? ( media-libs/libwebp:= )
103a108
> 		x11-libs/libXcursor
106d110
< 		x11-libs/libXcursor
113d116
< 		x11-libs/libXScrnSaver
114a118
> 		x11-libs/libXScrnSaver
126c130
< 			xpresent? ( x11-libs/libXpresent )
---
> 		xpresent? ( x11-libs/libXpresent )
129d132
< 
132,140d134
< 
< 	sys-apps/dbus
< 	sys-apps/util-linux
< 	sys-libs/zlib
< 	virtual/jpeg:0=
< 
< 	!media-libs/elementary
< 	!media-plugins/emotion_generic_players
< 	!media-plugins/evas_generic_loaders
148a143,144
> RDEPEND="${COMMON_DEPEND}"
> 
152c148
< 	# Remove sleep command that forces user to read warnings about their configuration
---
> 	# Remove sleep command that forces user to read warnings about their configuration.
171,175d166
< 	if use ssl && use gnutls ; then
< 		einfo "You enabled both USE=ssl and USE=gnutls, but only one can be used;"
< 		einfo "gnutls has been selected for you."
< 	fi
< 
307,309c298
< 	if ! use static-libs ; then
< 		find "${D}" -name '*.la' -delete || die
< 	fi
---
> 	find "${D}" -name '*.la' -delete || die
```

Pushing and rebasing in a minute.

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,309 @@
+	xim? ( X )
+"
+
+COMMON_DEPEND="
```
Not quite. ;) *this* should be RDEPEND (and move below), and inherit COMMON_DEPEND

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,309 @@
+	!media-plugins/evas_generic_loaders
+"
+
+RDEPEND="
```
this should be COMMON_DEPEND

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,309 @@
+"
+
+DEPEND="
+	${RDEPEND}
```
inherit COMMON_DEPEND instead.

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,309 @@
+	doc? ( app-doc/doxygen )
+"
+
+RDEPEND="${COMMON_DEPEND}"
```
drop this

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,309 @@
+IUSE="avahi +bmp dds connman debug doc drm +eet egl examples fbcon +fontconfig fribidi gif gles glib gnutls gstreamer harfbuzz hyphen +ico ibus ivi jpeg2k libuv luajit neon nls opengl ssl pdf physics pixman postscript +ppm +psd pulseaudio raw scim sdl sound static-libs svg systemd tga tiff tslib unwind v4l valgrind vlc vnc wayland webp X xcf xim xine xpresent xpm"
+
+REQUIRED_USE="
+	?? ( gnutls ssl )
```
this has gone too far. the solution before was not wrong, the warning was just bogus, and it is not necessary to make any `REQUIRED_USE` definitions for it. It is similar to the ffmpeg vs. libav situation, just more convoluted. `USE=ssl` means ssl support is entirely optional. Below that, if `gnutls` is disabled, list other options such as `libressl`. So please add back the previous construct in COMMON_DEPEND:
```
	ssl? (
		gnutls? ( >=net-libs/gnutls-3.3.6 )
		!gnutls? (
			!libressl? ( dev-libs/openssl:0= )
			libressl? ( dev-libs/libressl:0= )
		)
	)
```
(minus `libressl` if it really does not work)

juippis commented:
Sorted COMMON_DEPEND, DEPEND and RDEPEND. I think I got it right, as you wanted. 

Restored libressl IUSE and the construct for ssl. Also restored ssl-related stuff into REQUIRED_USE.

a17r commented:
> Also restored ssl-related stuff into REQUIRED_USE.
Please drop that. :)

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,314 @@
+
+	else
+		myconf+=( --with-opengl=none )
+
```
superfluous empty line

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,314 @@
+
+	elif use drm && use wayland ; then
+		myconf+=( --with-opengl=es )
+
```
superfluous empty line

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,314 @@
+
+	elif use egl ; then
+		myconf+=( --with-opengl=es )
+
```
superfluous empty line

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,314 @@
+
+	if use opengl ; then
+		myconf+=( --with-opengl=full )
+
```
superfluous empty line

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,314 @@
+		x11-libs/libXrender
+		x11-libs/libXtst
+		x11-libs/libXScrnSaver
+
```
superfluous empty line

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,314 @@
+			x11-libs/libXrender
+			virtual/opengl
+		)
+
```
superfluous empty line

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,314 @@
+			x11-libs/libX11
+			x11-libs/libXrender
+			virtual/opengl
+		xpresent? ( x11-libs/libXpresent )
```
indentation

a17r commented on dev-libs/efl/efl-1.20.7-r1.ebuild:
```diff
@@ -0,0 +1,314 @@
+	sys-apps/util-linux
+	sys-libs/zlib:=
+	virtual/jpeg:0=
+
```
superfluous empty line

juippis commented:
All done I think :)
